DatadogLogs/Sources/Logger.swift

```diff
@@ -0,0 +1,253 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// Log levels ordered by their severity, with `.debug` being the least severe and
+/// `.critical` being the most severe.
+public enum LogLevel: Int, Codable {
+    case debug
+    case info
+    case notice
+    case warn
+    case error
+    case critical
+}
+
+/// Datadog Logger.
+///
+/// Usage:
+///
+///     import DatadogLogs
+///
+///     // Initialise the Logs module
+///
+///     // logger reference
+///     var logger = DatadogLogger.builder.build()
+public protocol Logger {
```

Comment:
Yes, this `Logger` won't be used as a type, only `DatadogLogger`.

Reply:
Even then, the problem will be still there:
```swift
import DatadogLogs

let logger: Logger
```
☝️ It could be still ambiguous because `Logger` could either mean `DatadogLogs.Logger` protocol or `Logger` struct/enum/protocol/class defined in app module. I think there is no other way around than changing the name if we want to keep it in public interface. `DDLogger` could be a simple fix there - we could tackle it in separate JIRA task. WDYT?